addons/website/static/src/interactions/cookies/cookies_bar.js

```diff
@@ -0,0 +1,132 @@
+import { cookie } from "@web/core/browser/cookie";
+import { _t } from "@web/core/l10n/translation";
+import { registry } from "@web/core/registry";
+import { isVisible } from "@web/core/utils/ui";
+import { onceAllImagesLoaded } from "@website/interactions/utils";
+// import { setUtmsHtmlDataset } from "@website/js/content/inject_dom";
+import { Popup } from "@website/snippets/s_popup/000";
+
+// Extending the Popup class with cookiebar functionality.
+// This allows for refusing optional cookies for now and can be
+// extended to picking which cookies categories are accepted.
+export class CookiesBar extends Popup {
+    static selector = "#website_cookies_bar";
+    dynamicContent = Object.assign(this.dynamicContent, {
+        "#cookies-consent-essential, #cookies-consent-all": {
+            "t-on-click": this.onAcceptClick,
+        },
+    });
+
+    start() {
+        super.start();
+        this.addListener(
+            this.services.website_cookies.bus,
+            "show_cookies_bar",
+            this.onShowCookiesBar
+        );
+    }
+
+    async showPopup() {
+        super.showPopup();
+        const policyLinkEl = this.el.querySelector(".o_cookies_bar_text_policy");
+        if (policyLinkEl && window.location.pathname === new URL(policyLinkEl.href).pathname) {
+            this.toggleEl = wUtils.cloneContentEls(`
+            <button class="o_cookies_bar_toggle btn btn-info btn-sm rounded-circle d-flex gap-2 align-items-center position-fixed pe-auto">
+                <i class="fa fa-eye" alt="" aria-hidden="true"></i> <span class="o_cookies_bar_toggle_label"></span>
+            </button>
+            `).firstElementChild;
+            this.insert(this.toggleEl, this.el, "beforebegin");
+            await this.toggleCookiesBar();
+            this.addListener(this.toggleEl, "click", this.toggleCookiesBar);
```

Comment:
Same question here - but maybe that would be overkill.

Reply:
I hesitated, but I think it's a good idea indeed, even though that takes more lines of code. It makes it clearer what does what, and makes it easier to adapt to other needs. (specifically, while I was trying to make it work, I slightly tweaked the existing code to be able to reopen the cookies bar even after giving consent, which is a feature that was missing to be GDPR-compliant. For now it's only on the policy page, but it could easily be added on a link on a footer.)